### PR TITLE
Be equally forgiving of leading & trailing whitespace in number parsing

### PR DESCRIPTION
**Pull request by Tetralux**
Follow-up from #20588.

Before #20588, the parsing code for integers was forgiving of leading and trailing whitespace.
The boolean parsing code was not.

After #20588, the integer parsing code _was still_ forgiving of whitespace.
The boolean parsing code was _still not_.

This PR makes all three _forgiving_ of _all_ leading and trailing whitespace, as determined by `isspace`.

cc @tkelman

### CODE REVIEW

File: test/parse.jl [hunk 1]
```diff
@@ -243,6 +243,36 @@ for T in vcat(subtypes(Signed), subtypes(Unsigned))
...
+
+    # Test that the entire input string appears in error messages
+    let s = "     false    true     "
+        result = @test_throws ArgumentError get(Base.tryparse_internal(Bool, s, start(s), endof(s), 0, true))
+        @test result.value.msg == "invalid Bool representation: $(repr(s))"
+    end
+
+    # Test that leading and trailing whitespace is ignored.
+    for v in (1, 2, 3)
+         @test parse(Int, "    $v"    ) == v
```
Comment (tkelman) [inline test/parse.jl]: over indented by 1

**Comment by tkelman** [comment]
@nanosoldier `runbenchmarks(ALL, vs = ":master")`

**Comment by H-225** [comment]
@tkelman Oops. Might have inadvertently cancelled Nanosoldier by pushing?

**Comment by KristofferC** [comment]
Nope. It will post when done.

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/20606#issuecomment-279615479) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/d54fdb5ff479ac4e754a74e35adfcee7363c76cb/a46c9e3_vs_388435e/report.md). cc @jrevels

**Comment by H-225** [comment]
Hmmm...
```
["array","index", ("sumvector_view","SubArray{Float32,2,Array{Float32,3},Tuple{Int64,Base.Slice{Base.OneTo{Int64}},Base.Slice{Base.OneTo{Int64}}},true}")]
```
I'm not sure I understand what this means. Something about a slice? :/
Can one of you point me to where I can see this benchmark?

**Comment by StefanKarpinski** [comment]
That seems unrelated since we're definitely not doing any parsing during view indexing.

**Comment by H-225** [comment]
@StefanKarpinski That's what was confusing me.
RTM?

**Comment by H-225** [comment]
Thank you. 😄

**Pull request merged**